feat: Support __typename on all objects

glittershark opened the pull request:
Support a __typename introspection field on all queried objects. When requested,
it returns a string representing the name of the object type being queried.

glittershark committed:
feat: Support __typename on all objects

Support a __typename introspection field on all queried objects. When requested,
it returns a string representing the name of the object type being queried.

russmatney reviewed (approved):
LGTM.

That mortgage test always gets me because my brain goes to a totally different context.

russmatney committed:
feat: impl At and Ixed for ordered map, add lens

These make it possible to use lenses over orderedMaps.

```
-- | Looks up the value at the passed key in the passed object.
--
(.:) :: FromValue a => GQLValue.Object -> GQLValue.Name -> Either Text a
(Object' obj) .: key =
   case obj ^. at key of
     Nothing -> Left $ "`" <> tshow key <> "` field missing."
     Just v  -> fromValue v

-- | Similar to '(.:)', but returns a 'Maybe a'.
-- 'Right Nothing' means the field was not found.
-- Errors parsing to the desired type are returned as 'Left Text'.
(.:?) :: FromValue a => GQLValue.Object -> GQLValue.Name -> Either Text (Maybe a)
(Object' obj) .:? key =
   case obj ^. at key of
     Nothing -> Right Nothing
     Just v  -> fromValue v
```

Should move these into graphql-api at some point.
